frontend/lib/src/components/widgets/DateInput/DateInput.tsx

```diff
@@ -74,360 +65,320 @@ function datesToStrings(dates: Date[]): string[] {
-    const maxDate = this.getMaxDate()
-    const clearable = element.default.length === 0 && !disabled
-
-    // We need to extract the mask and format (date-fns notation) from the provided format string
-    // The user configured date format is based on the momentJS notation and is only allowed to contain
-    // one of YYYY/MM/DD, DD/MM/YYYY, or MM/DD/YYYY" and can also use a period (.) or hyphen (-) as separators.
-
-    // We need to convert the provided format into a mask supported by the Baseweb datepicker
-    // Thereby, we need to replace all letters with 9s which refers to any number.
-    const dateMask = element.format.replaceAll(/[a-zA-Z]/g, "9")
-    // The Baseweb datepicker supports the date-fns notation for date formatting which is
-    // slightly different from the momentJS notation. Therefore, we need to
-    // convert the provided format into the date-fns notation:
-    const dateFormat = element.format.replaceAll("Y", "y").replaceAll("D", "d")
-
-    // Manage our form-clear event handler.
-    this.formClearHelper.manageFormClearListener(
-      widgetMgr,
-      element.formId,
-      this.onFormCleared
-    )
+        })
+      } else {
+        newValue.push(date)
+      }
 
-    return (
-      <div className="stDateInput" data-testid="stDateInput" style={style}>
-        <WidgetLabel
-          label={element.label}
-          disabled={disabled}
-          labelVisibility={labelVisibilityProtoValueToEnum(
-            element.labelVisibility?.value
-          )}
-        >
-          {element.help && (
-            <StyledWidgetLabelHelp>
-              <TooltipIcon
-                content={element.help}
-                placement={Placement.TOP_RIGHT}
-              />
-            </StyledWidgetLabelHelp>
-          )}
-        </WidgetLabel>
-        <UIDatePicker
-          density={DENSITY.high}
-          formatString={dateFormat}
-          mask={isRange ? `${dateMask} – ${dateMask}` : dateMask}
-          placeholder={
-            isRange ? `${element.format} – ${element.format}` : element.format
-          }
-          disabled={disabled}
-          onChange={this.handleChange}
-          onClose={this.handleClose}
-          overrides={{
-            Popover: {
-              props: {
-                placement: PLACEMENT.bottomLeft,
-                overrides: {
-                  Body: {
-                    style: {
-                      border: `${theme.sizes.borderWidth} solid ${colors.borderColor}`,
-                    },
+      setValueWithSource({ value: newValue, fromUi: true })
+      setIsEmpty(!newValue)
+    },
+    [setValueWithSource]
+  )
+
+  const handleClose = useCallback((): void => {
+    if (!isEmpty) return
+
+    const newValue = stringsToDates(element.default)
+    setValueWithSource({ value: newValue, fromUi: true })
+    setIsEmpty(!newValue)
+  }, [isEmpty, element, setValueWithSource])
+
+  return (
+    <div className="stDateInput" data-testid="stDateInput" style={style}>
```

Comment:
nit (codebase style): instead of setting `style={style}` we should probably use a `styled-components.ts` file here and define the div there, because we usually avoid inline styling